src/Store.cpp

```diff
@@ -14,92 +14,210 @@ const unsigned short bombCost = 100;
 const unsigned short potionCost = 150;
 const unsigned short whetstoneCost = 20;
 
-//Messages
+// Messages
+const string notEnoughItemsMessage = " Insufficient Items !";
 const string lowCoinsMessage = " Insufficient Coins !";
 const string invalidChoice = " Invalid Choice !";
 
 void Store::StoreFront(Player* _Player) {
+	// Displays store menu and allows player to buy/sell items.
 
 	unsigned short choice = 0;
-	unsigned short coins;
+
+	bool buyItems = false;
+	bool isValidChoice = false;
 
 	do
 	{
 		ClearScreen();
-		coins = _Player->GetCoins();
 		cout << endl << endl;
 		_Player->DisplayInventory();
 		cout << endl << " Jeremy's Tools Store" << endl;
 		cout << " \"" << random_greeting() << "\"" << endl << endl;
+		cout << " ---------------------------------- " << endl << endl;
+		cout << " 1) Buy" << endl;
+		cout << " 2) Sell" << endl << endl;
+		cout << " 0) Exit" << endl << endl;
+		cout << " ---------------------------------- " << endl << endl;
+		cout << " What would you like to do today? " << endl << endl << " ";
+		choice = input();
+		cout << endl;
+
+		isValidChoice = true;
+		switch (choice)
```

Comment:
While this implementation works, I think it's better to separate buying/selling into two separate functions, rather than combining them here. This implementation assumes that whatever we buy can be sold, at the same price - what if in the future, the store buys back items at a lower price, or if the store sells unsellable items? So, IMO the buy/sell logic should be decoupled.

Reply:
I separated the buy logic from the sell logic in 1d9bb14c7be04b9eb871183b97c1962a3f17e5da and allowed both options to specify their own buyable/sellable items (including price and amount).